### DISCUSSION TITLE
采用PP-OCRv3 det文字检测 教师模型dml 训练的问题

### DISCUSSION THREAD

**Discussion by tairen99** [PaddlePaddle/PaddleOCR#12292] [Q&A]
Hi all and @WenmuZhou 

首先感谢你们提供了这个良好的OCR平台。

- 系统环境/System Environment：x86_64 Ubuntu 20.04.6 LTS 
- 版本号/Version：Paddle：2.6.0    PaddleOCR：2.7 
- 问题相关组件/Related components：ch_PP-OCRv3_det_cml.yml

根据 [文本检测模型训练](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_ch/PPOCRv3_det_train.md)，我想尝试使用 ch_PP-OCRv3_det_dml.yml 进行教师检测模型的训练。我首先想用 ICIDAR2015 的数据做训练测试，看看教师模型效果咋样；我测试了三种不同的YML配置文件：

第一种：将   "Global” 中的pretrained_model: ./pretrained_models/db/MobileNetV3_large_x0_5_pretrained 屏蔽了，因为可以避免导入阶段的各种警告，具体情况如下：
  ```
Global:
    use_gpu: true
    epoch_num: 1200
    log_smooth_window: 20
    print_batch_step: 2
    save_model_dir: ./output/db_icidar2015/
    save_epoch_step: 40
    # evaluation is run every 5000 iterations after the 4000th iteration
    eval_batch_step: [0, 500]
    cal_metric_during_train: False
    # pretrained_model: ./pretrained_models/db/MobileNetV3_large_x0_5_pretrained
    checkpoints:
    save_inference_dir:
    use_visualdl: False
    infer_img: doc/imgs_en/img_10.jpg
    save_res_path: ./output/det_db/predicts_db.txt
  Architecture:
    name: DistillationModel
    algorithm: Distillation
    model_type: det
    Models:
      Student:
        pretrained: ./pretrained_models/db/ResNet50_vd_ssld_pretrained
        freeze_params: false 
        return_all_feats: false
        model_type: det
        algorithm: DB
        Backbone:
          name: ResNet_vd
          in_channels: 3
          layers: 50
        Neck:
          name: LKPAN
          out_channels: 256
        Head:
          name: DBHead
          kernel_list: [7,2,2]
          k: 50
      Student2:
        pretrained: ./pretrained_models/db/ResNet50_vd_ssld_pretrained
        freeze_params: false 
        return_all_feats: false
        model_type: det
        algorithm: DB
        Backbone:
          name: ResNet_vd
          in_channels: 3
          layers: 50
        Neck:
          name: LKPAN
          out_channels: 256
        Head:
          name: DBHead
          kernel_list: [7,2,2]
          k: 50
  Loss:
    name: CombinedLoss
    loss_config_list:
    - DistillationDMLLoss:
        model_name_pairs:
        - ["Student", "Student2"]
        maps_name: "thrink_maps"
        weight: 1.0
        # act: None
        model_name_pairs: ["Student", "Student2"]
        key: maps
    - DistillationDBLoss:
        weight: 1.0
        model_name_list: ["Student", "Student2"]
        # key: maps
        name: DBLoss
        balance_loss: true
        main_loss_type: DiceLoss
        alpha: 5
        beta: 10
        ohem_ratio: 3
  Optimizer:
    name: Adam
    beta1: 0.9
    beta2: 0.999
    lr:
      name: Cosine
      learning_rate: 0.001
      warmup_epoch: 2
    regularizer:
      name: 'L2'
      factor: 0
  PostProcess:
    name: DistillationDBPostProcess
    model_name: ["Student", "Student2"]
    key: head_out
    thresh: 0.3
    box_thresh: 0.6
    max_candidates: 2000
    unclip_ratio: 1.5
  
  Metric:
    name: DistillationMetric
    base_metric_name: DetMetric
    main_indicator: hmean
    key: "Student"
  
  Train:
    dataset:
      name: SimpleDataSet
      data_dir: ./train_data/icdar2015/
      label_file_list:
        - ./train_data/icdar2015/train_icdar2015_label.txt
      ratio_list: [1.0]
      transforms:
        - DecodeImage: # load image
            img_mode: BGR
            channel_first: False
        - DetLabelEncode: # Class handling label
        - CopyPaste:
        - IaaAugment:
            augmenter_args:
              - { 'type': Fliplr, 'args': { 'p': 0.5 } }
              - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
              - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
        - EastRandomCropData:
            size: [960, 960]
            max_tries: 50
            keep_ratio: true
        - MakeBorderMap:
            shrink_ratio: 0.4
            thresh_min: 0.3
            thresh_max: 0.7
        - MakeShrinkMap:
            shrink_ratio: 0.4
            min_text_size: 8
        - NormalizeImage:
            scale: 1./255.
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: 'hwc'
        - ToCHWImage:
        - KeepKeys:
            keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
    loader:
      shuffle: True
      drop_last: False
      batch_size_per_card: 2
      num_workers: 1
  
  Eval:
    dataset:
      name: SimpleDataSet
      data_dir: ./train_data/icdar2015/
      label_file_list:
        - ./train_data/icdar2015/test_icdar2015_label.txt
      transforms:
        - DecodeImage: # load image
            img_mode: BGR
            channel_first: False
        - DetLabelEncode: # Class handling label
        - DetResizeForTest:
  #           image_shape: [736, 1280]
        - NormalizeImage:
            scale: 1./255.
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: 'hwc'
        - ToCHWImage:
        - KeepKeys:
            keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
    loader:
      shuffle: False
      drop_last: False
      batch_size_per_card: 1 # must be 1
      num_workers: 2
```
第二种情况，将 "Global” 中的pretrained_model: ./pretrained_models/db/MobileNetV3_large_x0_5_pretrained 还原
第三种情况，将所有的 student中的 pretrained: ./pretrained_models/db/ResNet50_vd_ssld_pretrained，
                                      更换成       pretrained: ./pretrained_models/db/MobileNetV3_large_x0_5_pretrained；因为参考了 [帖子](https://github.com/PaddlePaddle/PaddleOCR/issues/9053)，怀疑训练数据不够

可是所有这三种情况，我都训练了600 epoch，best metric hmean 都小于 0.72，无法再提高。不知道什么原因，也不知道哪里出现了问题。
恳请各位指点，支招，谢谢大家！
